src/workers/prometheus.rs

```diff
@@ -0,0 +1,59 @@
+use std::collections::HashMap;
+
+use tokio::{
+    io::AsyncWriteExt,
+    net::{TcpListener, TcpStream},
+    task::JoinHandle,
+};
+
+use super::MetricsCapture;
+
+#[derive(Clone)]
+pub struct PrometheusSync {
+    metrics_capture: MetricsCapture,
+}
+
+impl PrometheusSync {
+    pub fn new(metrics_capture: MetricsCapture) -> Self {
+        Self { metrics_capture }
+    }
+
+    fn create_response(payload: &str) -> String {
+        format!(
+            "HTTP/1.1 200 OK\r\nContent-Length: {}\r\n\r\n{}",
+            payload.len(),
+            payload
+        )
+    }
+
+    async fn handle_stream(&self, stream: &mut TcpStream) -> anyhow::Result<()> {
+        let metrics = self.metrics_capture.get_metrics().await;
+        let metrics = serde_prometheus::to_string(&metrics, None, HashMap::new())?;
```

Comment:
Please add "literpc" as a prefix so we can namespace metrics https://github.com/w4/serde_prometheus/blob/master/src/lib.rs#L89

Reply:
Done